### PR TITLE
API for adding bulk songs

### PR DESCRIPTION
**Pull request by krishrustagi**

### CODE REVIEW

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 1]
```diff
@@ -37,12 +41,33 @@
...
 
+    private static final int BULK_SONGS_LIMIT = 50;
+
     private final SongsDao songsDao;
     private final UserDao userDao;
     private final UserSongMappingDao userSongMappingDao;
 
     @Override
-    public ResponseEntity<String> addSong(MultipartFile songFile) throws IOException {
+    public ResponseEntity<String> addSongs(List<MultipartFile> songFiles) throws IOException {
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: yeh function bahut bada ho rha hai isko split krde bs

File: src/main/java/com/emotunes/emotunes/service/AdminService.java [hunk 2]
```diff
@@ -5,10 +5,11 @@
 import org.springframework.web.multipart.MultipartFile;
 
 import java.io.IOException;
+import java.util.List;
 
 public interface AdminService {
 
-    ResponseEntity<String> addSong(MultipartFile songFile) throws IOException;
+    ResponseEntity<String> addSongs(List<MultipartFile> songFiles) throws IOException;
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/AdminService.java]: Should return String Not Response Entity, handle response entity in controller

Comment (krishrustagi) [reply src/main/java/com/emotunes/emotunes/service/AdminService.java]: response entity returning for internal server error ke liye bhi

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 3]
```diff
@@ -38,12 +42,33 @@
...
+
     private final SongsDao songsDao;
     private final UserDao userDao;
     private final UserSongMappingDao userSongMappingDao;
 
     @Override
-    public ResponseEntity<String> addSong(MultipartFile songFile) throws IOException {
+    public ResponseEntity<String> addSongs(List<MultipartFile> songFiles) { // todo: use multithreading
+        if (songFiles.size() > BULK_SONGS_LIMIT) {
+            return ResponseEntity.internalServerError().body("Maximum 50 files at a time allowed!");
```
Comment (krishrustagi) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: Here @SarthakSavasil

Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: Bad data error jesa kuch dena chaiye! Internal server ki jagah kuch aur predefined error bhi honge

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 4]
```diff
@@ -38,12 +42,33 @@
...
+        } else {
+            songFiles.forEach(songFile -> {
+                try {
+                    addSong(songFile);
+                } catch (IOException | CannotReadException | TagException | InvalidAudioFrameException |
+                         ReadOnlyFileException e) {
+                    log.error("Error while adding song {}", songFile, e);
+                    throw new RuntimeException();
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: RunTime bhi 500 error create krta

**Review by SarthakSavasil — CHANGES_REQUESTED** [review]

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java (hunk 4 above)
Comment (krishrustagi) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: But comment body nhi deta

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java (hunk 3 above)
Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: throw exception

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java (hunk 4 above)
Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: 500 nhi jana chaiye.. koi aur error ka option nhi hai?

Comment (krishrustagi) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: 500 hi jaana chahiye na

Comment (krishrustagi) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: 500 gaya toh jitne hue utne db mein gaye...

Comment (krishrustagi) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: baaki error search krke bhej denge.. varna logs check krte raho

Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: Aur yaha throw hona hi nhi chaiye

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java [hunk 5]
```diff
@@ -38,21 +42,46 @@
...
+
     private final SongsDao songsDao;
     private final UserDao userDao;
     private final UserSongMappingDao userSongMappingDao;
 
     @Override
-    public ResponseEntity<String> addSong(MultipartFile songFile) throws IOException {
+    public String addSongs(List<MultipartFile> songFiles) { // todo: use multithreading
+        if (songFiles.size() > BULK_SONGS_LIMIT) {
+            throw new RuntimeException("Maximum 50 files at a time allowed!");
```
Comment (SarthakSavasil) [inline src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: IllegalArgumentException

File: src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java (hunk 4 above)
Comment (SarthakSavasil) [reply src/main/java/com/emotunes/emotunes/service/impl/AdminServiceImpl.java]: throw krne ke baad loop continue nhi hoga na

**Review by SarthakSavasil — APPROVED** [review]

**Pull request merged**